### PR TITLE
Introduce an optional `zero_threshold` field to `ExponentialHistogramDataPoint`

### LINKED ISSUE #440: Add optional `zero_threshold` to ExponentialHistogramDataPoint.
It is part of the spec, see [Zero Count and Zero Threshold](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md?plain=1#L622-L640). 
The spec change was proposed in [opentelemetry-specification#2665](https://github.com/open-telemetry/opentelemetry-specification/pull/2665).

`zero_threshold` will be used to convert the OTel Exponential Histograms to Prometheus Native Histograms, see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/16207

### PR DESCRIPTION
**Pull request by kovrus**
Introduce an optional `zero_threshold` field to `ExponentialHistogramDataPoint`.
See [Zero Count and Zero Threshold](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md?plain=1#L622-L640).

Resolves #440

### CODE REVIEW

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 1]
```diff
@@ -561,6 +561,12 @@ message ExponentialHistogramDataPoint {
 
   // max is the maximum value over (start_time, end_time].
   optional double max = 13;
+
+  // ZeroTolerance may be optionally set to convey the width of the
+  // zero region. When ZeroTolerance is unset or 0, this bucket stores values
```
Comment (Aneurysm9) [inline opentelemetry/proto/metrics/v1/metrics.proto:566]:
```suggestion
  // ZeroThreshold may be optionally set to convey the width of the
  // zero region. When ZeroThreshold is unset or 0, this bucket stores values
```

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 2]
```diff
@@ -561,6 +561,12 @@ message ExponentialHistogramDataPoint {
 
   // max is the maximum value over (start_time, end_time].
   optional double max = 13;
+
+  // ZeroThreshold may be optionally set to convey the width of the
+  // zero region. When ZeroThreshold is unset or 0, this bucket stores values
```
Comment (gouthamve) [inline opentelemetry/proto/metrics/v1/metrics.proto:566]: `this bucket` is mileading. I think you mean the `zero_count`?

Comment (kovrus) [reply opentelemetry/proto/metrics/v1/metrics.proto:566]: maybe `this bucket` -> `zero count bucket`?

**Comment by kovrus** [comment]
cc: @dashpole

File: opentelemetry/proto/metrics/v1/metrics.proto [hunk 3]
```diff
@@ -561,6 +561,13 @@ message ExponentialHistogramDataPoint {
 
   // max is the maximum value over (start_time, end_time].
   optional double max = 13;
+
+  // ZeroThreshold may be optionally set to convey the width of the
+  // zero region.
+  // When ZeroThreshold is unset or 0, zero count bucket stores
+  // values that cannot be expressed using the standard exponential formula as
+  // well as values that have been rounded to zero.
```
Comment (jmacd) [inline opentelemetry/proto/metrics/v1/metrics.proto:569]: Please add "The zero region is defined as the closed interval [-ZeroThreshold, ZeroThreshold]." That is to say that the boundary is inclusive of the threshold on both sides.

**Review by jack-berg — APPROVED** [review]

**Review by reyang — APPROVED** [review]

**Comment by kovrus** [comment]
@open-telemetry/specs-approvers can we merge that?

**Comment by reyang** [comment]
> Hey @open-telemetry/specs-approvers, can you merge that, please?

Please get more folks to review the PR.

**Comment by kovrus** [comment]
@reyang any suggestions on whom should I ask to review this PR?

**Comment by reyang** [comment]
> @reyang any suggestions on whom should I ask to review this PR?

@open-telemetry/specs-metrics-approvers @dashpole @gouthamve @pirgeo

**Review by dashpole — APPROVED** [review]

**Pull request merged**